pkg/ingester/activeseries/active_series.go

```diff
@@ -114,14 +114,14 @@ func (c *ActiveSeries) ConfigDiffers(ctCfg asmodel.CustomTrackersConfig, caCfg *
 	return ctCfg.String() != c.matchers.Config().String() || caCfg != c.cat
 }
 
-func (c *ActiveSeries) ReloadMatchers(asm *asmodel.Matchers, now time.Time) {
+func (c *ActiveSeries) ReloadMatchersAndTrackers(asm *asmodel.Matchers, cat *costattribution.ActiveSeriesTracker, now time.Time) {
 	c.configMutex.Lock()
 	defer c.configMutex.Unlock()
-
 	for i := 0; i < numStripes; i++ {
-		c.stripes[i].reinitialize(asm, &c.deleted, c.cat)
+		c.stripes[i].reinitialize(asm, &c.deleted, cat)
 	}
 	c.matchers = asm
+	c.cat = cat
```

Comment:
nit: Can you add a note on the `cat` field of this struct to indicate it is protected by the `configMutex`?

Reply:
I'd say that instead of the comment, it should be moved one line to be covered by the _mutex hat_.

I agree that with current arrangement it's not clear whether it's covered by the `configMutex`

Reply:
added comments + move cat cerca de mutex